### ISSUE TITLE
Cannot capture any stream

### ISSUE THREAD

**Issue by bananarama42** [rytsikau/ee.Yrewind#24]
As of today I seem to be unable to capture any stream. I have downloaded the latest version with no luck. 
It gets stuck at "getting live stream info...", counts down a few times and then exits. 

I have tested with several different url's but keep getting the same outcome.

**Comment by duvi** [comment]
I'm having the same issue, even with the example stream from run.bat

**Comment by btom1970** [comment]
hello!
the program doesn't work from last night! (2021.06.18 03.00AM)
Previous days all working perfect, 2021.06.17 evening too...
thanks!

checked:
- latest (2021.06.15-ös) version
- older version (2021.01)
- with other ffmpeg
- with "http" and "https"
- with "youtube id"
- with full link
- many links
- many time format
With no luck...

**Comment by rytsikau** [comment]
Hi!
Obviously, these are some changes on the YouTube side. I'll post the fixed version later

**Comment by brolnickij** [comment]
same problem

**Comment by rytsikau** [comment]
Program has been updated. It now requires installed Google Chrome or MS Edge

**Comment by nanahsu315** [comment]
> Program has been updated. It now requires installed Google Chrome or MS Edge

thank you!!!

**Comment by bananarama42** [comment]
Thank you for fixing 👍 Was able to get a brief test clip, but now I keep getting error 9229 even with the first sample url.

**Comment by brolnickij** [comment]
> Program has been updated. It now requires installed Google Chrome or MS Edge

awesome, thank you!

**Comment by btom1970** [comment]
I get 9229 error.
rytsikau, can You post a fixed version? Thanks!

**Comment by rytsikau** [comment]
This error means that your browser does not load the html page with the stream to the end (in headless mode, of course). There can be many reasons - unstable Internet connection, incorrect installation of the browser, buggy add-ons in the browser, etc. The only thing I can do on my end is to add the option to explicitly select which browser to use to get the html. But I don’t have time for this yet, and it will not appear soon.

**Comment by rytsikau** [comment]
Try also to check if autoplay is enabled on Youtube page

![20210622-080124](https://user-images.githubusercontent.com/47730840/122866252-4dace700-d330-11eb-92ef-0e6b4b05210a.jpg)

**Comment by btom1970** [comment]
bananarama42, You have still the error message?

**Comment by bananarama42** [comment]
> bananarama42, You have still the error message?

Yes, sadly. I have no idea why I got a clip the first time, but after that it keeps failing permanently. Disabled all extensions, even tried a different computer.

**Comment by btom1970** [comment]
brolnickij, nanahsu315, duvi - Your program working?

**Comment by nanahsu315** [comment]
Yes, it works.
Two computers: 1.chrome &Edge 2.only Edge

yrewind -url="https://www.youtube.com/watch?v=XXXXX" -start=20210621:090000 -duration=60 -output='E:\20210621\20210621-0400'

The more troublesome thing is that on average 10 downloads will have 2 downloads incomplete, which has not appeared before

**Comment by btom1970** [comment]
nanahsu315, 

My program not working... 
getting live stream info...--> ERROR 9229: Cannot get live stream information (2021.06.22. 10:19:23)

You use Windows10 ?

My computer:
Google Chrome
Version: 91.0.4472.114 
Windows 7

**Comment by nanahsu315** [comment]
> nanahsu315,
> 
> My program not working...
> getting live stream info...--> ERROR 9229: Cannot get live stream information (2021.06.22. 10:19:23)
> 
> You use Windows10 ?
> 
> My computer:
> Google Chrome
> Version: 91.0.4472.114
> Windows 7

both :win10

chrome:91.0.4472.106
edge:91.0.864.54

I just tried a live stream that can't be embedded, and the message :
ERROR 9231: Cannot get live stream information (2021/06/22 下午 04:44:24)

Can the  live stream you download cannot be embedded?

**Comment by kettlerne** [comment]
I also get the 9229 error

Windows 10
Edge: 91.0.864.54

Tried with different streams (that can be embedded); autoplay off/on. Still getting the error

**Comment by nanahsu315** [comment]
Is there any antivirus software?
Both of my computers only use win10 built-in

**Comment by bananarama42** [comment]
nanahsu315 I turned my antivirus software off and that didn't help. Uninstalling it also didn't help.

**Comment by bananarama42** [comment]
> Is there any antivirus software?
> Both of my computers only use win10 built-in

nanahsu315 I turned my antivirus software off and that didn't help. Uninstalling it also didn't help.

**Comment by nanahsu315** [comment]
> > Is there any antivirus software?
> > Both of my computers only use win10 built-in
> 
> nanahsu315 I turned my antivirus software off and that didn't help. Uninstalling it also didn't help.

That's really weird,
I can use the program on both computers
Sorry for not being able to help

**Comment by bananarama42** [comment]
> > > Is there any antivirus software?
> > > Both of my computers only use win10 built-in
> > 
> > 
> > nanahsu315 I turned my antivirus software off and that didn't help. Uninstalling it also didn't help.
> 
> That's really weird,
> I can use the program on both computers
> Sorry for not being able to help

That's ok :) Thank you for offering some suggestions. The weird thing is that it worked once for me. So maybe there is something blocking it, but I have no idea what.

**Comment by btom1970** [comment]
Looks like for me and 2 others not working...
What is the reason...?

**Comment by btom1970** [comment]
- I uninstalled Edge and Chrome, reinstalled
- Windows safe mode
no luck

**Comment by Explorare** [comment]
Works for me with MS Edge. Did your network environment need proxy? Setting up proxy for this tool is a little tricky.

**Comment by bananarama42** [comment]
> Works for me with MS Edge. Did your network environment need proxy? Setting up proxy for this tool is a little tricky.

I didn't need a proxy before, so not sure why I would need one now.

**Comment by Explorare** [comment]
You don't need to set up the proxy for this tool if you could access YouTube directly. And is your MS Edge a 64bit version?

**Comment by bananarama42** [comment]
> You don't need to set up the proxy for this tool if you could access YouTube directly. And is your MS Edge a 64bit version?

Yes it is, although I assume it might pick Chrome first as that is also installed.

**Comment by bob-swede** [comment]
I get this with 21.063:
`C:\Programs\YRewind>yrewind.exe -url=https://youtu.be/wEcy1yBTNPQ -resolution=480 -duration=12 -start=T:1057
--> ee.Yrewind (started at 2021-06-23 11:11:22)
getting live stream info...
--> ERROR 9229: Cannot get live stream information (2021-06-23 11:12:13)`
My System: Windows 10 Pro x64, Chrome latest, Firefox Latest never using Edge.
Have used this program for a long time, recently stopped working.

**Comment by btom1970** [comment]
Explorare,
Do You haven't installed Chrome? Only Edge?

bob-swede
I (and someones) have the same error. We are searching for a solution...

**Comment by Explorare** [comment]
I just tested it on two computers and both work fine, one with Firefox Dev + Chrome + MS Edge, the other with FF Dev + MS Edge, both system is Windows 10 Enterprise 64bit, and all of those browsers are 64bit. 

I did encounter similar error in the past, when I messed up my proxy settings. My network environment requires a proxy to access YouTube, and I fixed the issue by using Proxifier to force the yrewind and ffmpeg processes go through my local proxy server, and a whitelist for prot 7799 which is used for local communications between yrewind and ffmpeg. And I have also changed the system proxy settings to use my local proxy server, which affects the browser connections.

**Comment by btom1970** [comment]
WORKING NOW....!!! (but on my mother's laptop)
She is on the same network, Win7, Edge+Chrome.

**Comment by btom1970** [comment]
Worked only one times... Downloaded 1 stream.
Still not working on my laptop, on my mother's laptop.
Like babanarama42, same situation.

**Comment by bob-swede** [comment]
I only tested with the Youtube URL in my post [MSNBC](https://youtu.be/wEcy1yBTNPQ) which is a stream of MSNBC news, but others have said that even the example URL in the download does not work...
I know that this worked before, that was how I tested the program.

**Comment by bananarama42** [comment]
I tried just about everything I could think of, but no luck so far.

**Comment by rytsikau** [comment]
Try also disable option "Use hardware acceleration when available" in browser

**Comment by bob-swede** [comment]
> Try also disable option "Use hardware acceleration when available" in browser

No change, except now I am getting this response (both with and without hardware accelereation):

yrewind.exe -url=https://youtu.be/wEcy1yBTNPQ -resolution=480 -duration=12 -start=T:0400
--> ee.Yrewind (started at 2021-06-24 08:34:32)
getting live stream info...
--> ERROR 9223: URL not found (2021-06-24 08:34:35)

And the URL is copied from the web browser playing the video (both Chrome and FireFox)

**Comment by bananarama42** [comment]
> Try also disable option "Use hardware acceleration when available" in browser

Gave this a try on two machines, but it didn't help, sorry :(

**Comment by btom1970** [comment]
> Try also disable option "Use hardware acceleration when available" in browser

I try this, but not working...

rytsikau, Yesterday worked for me - BUT ONLY ONCE - on my mother laptop. Second time, few minutes later, on same laptop it doesn't worked anymore... Nothing was changed on this laptop. Why?

**Comment by btom1970** [comment]
Wait... IT'S WORKING...!!!! (but takes some time)

**Comment by bananarama42** [comment]
> Wait... IT'S WORKING...!!!! (but takes some time)

You are right, all of a sudden it is working! How odd, because I didn't change anything since testing a few times earlier this morning.  I hope it is working for everyone else as well.

edit: it's a bit erratic, sometimes it works now, more often it fails.

**Comment by btom1970** [comment]
Me now.... not working anymore....
Try more?

**Comment by bananarama42** [comment]
> Me now.... not working anymore....
> Try more?

I'd say 9 times out of 10 it fails for me. Which honestly I don't understand.

**Comment by btom1970** [comment]
Me ONLY ONE download today and ONE yesterday...

**Comment by kettlerne** [comment]
So i tried different streams, Uninstalled Edge and installed Chrome (64bit), turned the windows antivirus software off, disabled the hardware acceleration. So far I've had no luck

**Comment by bananarama42** [comment]
So far I've managed to get a few clips today, but that was trial and error. I had to repeat it many many times to get one attempt working. And my last attempt not being successful at all.

**Comment by btom1970** [comment]
> Try also disable option "Use hardware acceleration when available" in browser

rytsikau, I see it is not working for most of people here. Sometimes working... You will post a new version of the program?
Approximately when?
thanks!

**Comment by Explorare** [comment]
It will spend a lot time for him to fix the issue which could not be reproduced in his dev environment. I think maybe providing more detailed error log would help with diagnosing the issue.

**Comment by computerfreak105** [comment]
Hi

I was gone for some time and now discovered the new version.
Would it be possible to make the program write log files we could attach to our reports?

I see, we have the option to include the path to cookies.
Should we set the path to where google chrom stores it’s cookies? It’s a file without extension in

"%localappdata%\Google\Chrome\User Data\Default\"

Or do we have to export them to a text file first in order to make it work for yrewind?


After a few manual downloads of ongoing streams I can say:


This channel here did a stream which couldn’t be saved:

channel: https://www.youtube.com/channel/UCEBib2DGF5Q25DwvHJyBFIg
  Stream url gave Error 9231: https://www.youtube.com/watch?v=tMI4bD6wX1Q

This channel here did a stream and the program didn’t release the file after the stream was finished:

channel: https://www.youtube.com/channel/UCltISp9vIGnVHyF6DdKmz-Q
 
stream url: https://www.youtube.com/watch?v=3QjeFAhnKVQ

Hope we can help to make it work for all again.

**Comment by bananarama42** [comment]
> It will spend a lot time for him to fix the issue which could not be reproduced in his dev environment. I think maybe providing more detailed error log would help with diagnosing the issue.

I wouldn't mind giving a more detailed error log, but the only thing I can offer at the moment is the error code.

My approach now is to copy the line about 10 times in the bat file and have it run. If I'm lucky one of them gets going. Sometimes I have to repeat this 5 times before it works. So it isn't that the application doesn't work at all in my case, but it is a bit of trial and error.

**Comment by rytsikau** [comment]
I really can't reproduce error 9229. All the possible reasons I wrote about (browser add-ons, hardware acceleration, autoplay button on Youtube page) are just speculations. By the way, one more thing - try disabling HTML5 player in browser.
Later I will make a version with the ability to report (do not ask when - I do not know)

**Comment by rytsikau** [comment]
To check the HTML5 assumption,
try to delete all YouTube cookies in the browser - after that it switches from HTML5 to regular Flash (at least for the first time)

**Comment by computerfreak105** [comment]
Ok, thanks, I deleted all history in chrome and disabled hardware accelleration. Are there any specific addons one should avoid? I have three addons installed:
1. AutoCaptcha solver, solves google’s recaptcha automatically in the background.

2. GetCookies, to export cookies to a file.

3. MalwareBytes browser guard

I understand this is a difficult task for you. So thanks for not giving up and all your work so far.

I figured out how the cookies integration works and will adjust my batch scripts.
Is it no problem to let different instances of yrewind access one cookies.txt file or should I copy it for all channels separately?
Same question goes for the ffmpeg.exe.
Until now I have a running batch scripts for every channel i monitor. in their folders. Every folder also includes yrewind.exe and ffmpeg.exe.
Is this the correct way?

**Comment by rytsikau** [comment]
computerfreak105, if you are talking about the '-cookie' parameter, then it does not depend in any way on cookies in the browser, or on hardware accelleration, or on addons. Better create a separate topic for this

**Comment by btom1970** [comment]
bananarama42, thanks for the "solution" - we must try many times - one will be successful

Working now for me. But I must try many times. One succesful... (this is a half solution is better than nothing)

**Comment by btom1970** [comment]
so, It is working now for me. But it must try 10-30 times, and 1x will be succesful. (with bat file)
rytsikau, It is a great program anyway. Thanks for it!

**Comment by bananarama42** [comment]
I have landed on error 9236 now. Maybe Youtube changed something again.

Thank you for all your hard work rytsikau. It is really appreciated. :)

**Comment by btom1970** [comment]
For me working now. (try 10-30 times - 1 succesful) I succesfully download the previous 6-7 days of my streams.

**Comment by btom1970** [comment]
Not working again from time interval "2021.06.25 - 2021.06.30" streams.
9236 error
Worked good (10-30 try) from time interval "2021.06.20 - 2021.06.25" streams.

rytsikau, what is 9236 error?

**Comment by bananarama42** [comment]
> Not working again from time interval "2021.06.25 - 2021.06.30" streams.
> 9236 error
> Worked good (10-30 try) from time interval "2021.06.20 - 2021.06.25" streams.
> 
> rytsikau, what is 9236 error?

I think that's when you have tried too many times. Youtube doesn't like that. At least I assume that's my problem at the moment. Whoops.

**Comment by btom1970** [comment]
> > Not working again from time interval "2021.06.25 - 2021.06.30" streams.
> > 9236 error
> > Worked good (10-30 try) from time interval "2021.06.20 - 2021.06.25" streams.
> > rytsikau, what is 9236 error?
> 
> I think that's when you have tried too many times. Youtube doesn't like that. At least I assume that's my problem at the moment. Whoops.

I trying with VPN now...
Error message is changing, I get multiple error message. I can download one partial stream, one times. Trying for few hours...

**Comment by anephew** [comment]
not working, same
getting live stream info...--> ERROR 9229: Cannot get live stream information

WIN 21H1 (19043.1052)
Microsoft Edge
Version 91.0.864.64 (Official build) (64-bit)

**Comment by btom1970** [comment]
Create a batch file and put the commands in there 10-30 times. One will download the stream. If not, try change the time in command a few minutes forward or rewind.

**Comment by btom1970** [comment]
rytsikau, I get this error... Why? Thanks! 

h:\yrewindnew3>yrewind -url='https://www.youtube.com/watch?v=_lejZpsJAOo' -start=20210719:1150 -duration=70
--> ee.Yrewind (started at 2021.07.20. 20:09:26)
getting live stream info...
    stream title        Gniazdo OstroÅ‚Ä™ka - Bociany na Å¼ywo "Osti i Lenka"
    resolutions         720,480,360,240,144
    output file         _lejZpsJAOo_20210719-115000_070m_0720p.mp4

--> ERROR 9411: Output file not created (2021.07.20. 20:09:44)

**Comment by rytsikau** [comment]
Hi
I currently only have access to slow mobile internet. Will check later

**Comment by btom1970** [comment]
Someone has this error?

ERROR 9411: Output file not created

**Comment by computerfreak105** [comment]
Yes, since maybe last week. I never had it earlier.

**Comment by rytsikau** [comment]
Try changing the output format. For example .ts, .avi etc. (perhaps it's an FFmpeg an error)

**Comment by computerfreak105** [comment]
I encountered it with .flv. No output format does really provide reliable downloads at the moment, as far I know.
I understand these are difficult times, probably especially in your area. Hope you’ll get faster internet soon.

Take care.

**Comment by bob-swede** [comment]
Tried to use this program today but I get this strange error:
```
yrewind.exe -url=https://youtu.be/DM-I6UugaFs -resolution=480 -duration=66 -start=T:0400
--> ee.Yrewind (started at 2021-08-16 08:07:48)
getting live stream info...
--> ERROR 9223: URL not found (2021-08-16 08:07:50)

The URL works just fine in both Firefox and Chrome, so what is the deal here?
```
I am using version 21.063

**Comment by rytsikau** [comment]
Hi! Did you tried using quotes with url?

**Comment by bob-swede** [comment]
No, I did not. Now testing this:
```
C:\Programs\YRewind>yrewind.exe -url="https://youtu.be/DM-I6UugaFs" -resolution=480 -duration=66 -start=T:0400
--> ee.Yrewind (started at 2021-08-16 10:48:33)
getting live stream info...
    stream title        Morning Joe - 8/13/21 msnbc
    resolutions         720,480,360,240,144
    output file         DM-I6UugaFs_20210816-040000_066m_0480p.mp4
```

Now there is no error message and the download seems to start, but it is very slow. Only 2.5x, used to be 20x or even more.

**Comment by bob-swede** [comment]
Video downloaded and is correct length, but after 3:46 min audio stops for the total length of the video! :(
And the video is marked INCOMPLETE in the name. Why?
When I play the URL in the web browser it is not muted.

**Comment by bob-swede** [comment]
Tested again and this time the video stops a few minutes into the file but audio continues...

**Comment by rightsaid** [comment]
Yrewind records only from 2 to 10 minutes of the ongoing stream now. Is there a fix?

**Comment by brolnickij** [comment]
https://github.com/Kethsar/ytarchive

**Comment by bob-swede** [comment]
I looked at:  https://github.com/Kethsar/ytarchive
According to the Readme.md of this "solution" it does not have an argument to set the start of the download.
Or maybe I have missed something?
Looks like it is always downloading from the start (i.e. archiving the whole stream), which could be 12 hours in the past. Not a viable replacement for ytrewind....
But it is good that it is with Python source so it can be running on Linux.

**Comment by rytsikau** [comment]
Hi!
Just wondering if this script works? As I understand, it uses an URL like https://www.youtube.com/get_video_info?video_id=
But this way seems to have stopped working a few months ago.

**Comment by bob-swede** [comment]
The URL should look like this example:
https://youtu.be/Zitz6XaHCxM
But it stopped working properly months ago (see above), probably because Youtube changed the way the videos are streamed and the maintainer of this program could not solve the problem.
I have reverted to using youtube-dl instead and I have set it up to record each hour separately in a list of at jobs on my Ubuntu server...
No way to rewind when using youtube-dl, but it does work to download the video stream.
Youtube-dl can also not be set to run for a specific time period so I had to use the timeout command to stop it after the wanted time.
This is my actual download command:
`COMMAND="youtube-dl -f $QUALITY -o $TARGETFILE $VIDEOURL"`
Where QUALITY is one of the available stream ID:s, I use QUALITY="94"
TARGETFILE is the output file like 2021-10-03_1200_input.mp4
VIDEOURL is the Youtube video URL as shown above or simiilar
This command is used to create an at job running at a specific time like so:
`eval "echo \"timeout --signal=2 62m ${COMMAND}\" | at 12:00"`
This creates the at job which runs at 12:00 for 62 minutes to download to the file 2021-10-03_1200_input.mp4
As long as the stream does not stop it creates a video lasting for the set timeout.
HTH
